rfhub2/cli/population.py

```diff
@@ -0,0 +1,175 @@
+        return self._is_library_file(file) or \
+               self._is_libdoc_file(file) or \
+               self._is_resource_file(file)
+
+    @staticmethod
+    def _is_library_file(file) -> bool:
+        return file.endswith(".py") and not file.endswith("__init__.py")
+
+    @staticmethod
+    def _is_libdoc_file(file) -> bool:
+        """Return true if an xml file looks like a libdoc file"""
+        # inefficient since we end up reading the file twice,
+        # but it's fast enough for our purposes, and prevents
+        # us from doing a full parse of files that are obviously
+        # not libdoc files
+        if file.lower().endswith(".xml"):
+            with open(file, "r") as f:
+                # read the first few lines; if we don't see
+                # what looks like libdoc data, return false
+                data = f.read(200)
+                index = data.lower().find("<keywordspec ")
+                if index > 0:
+                    return True
+        return False
+
+    @staticmethod
+    def _should_ignore(name) -> bool:
+        """Return True if a given library name should be ignored
+        This is necessary because not all files we find in the library
+        folder are libraries. I wish there was a public robot API
+        for "give me a list of installed libraries"...
+        """
+        filename, _ = os.path.splitext(name)
+        _name = filename.lower()
+        return (_name.startswith("deprecated") or
+                _name.startswith("_") or
+                _name in ("remote", "reserved", "dialogs", "dialogs_jy", "dialogs_py", "dialogs_ipy"))
+
+    @staticmethod
+    def _is_resource_file(file) -> bool:
+        """Return true if the file has a keyword table but not a testcase table"""
+        # inefficient since we end up reading the file twice,
+        # but it's fast enough for our purposes, and prevents
+        # us from doing a full parse of files that are obviously
+        # not robot files
+
+        if re.search(r'__init__.(txt|robot|html|tsv)$', file):
+            # These are initialize files, not resource files
+            return False
+
+        found_keyword_table = False
+        if os.path.splitext(file)[1].lower() in RESOURCE_PATTERNS:
+            with open(file, "r") as f:
+                data = f.read()
+                for match in re.finditer(r'^\*+\s*(Test Cases?|(?:User )?Keywords?)',
+                                         data, re.MULTILINE | re.IGNORECASE):
+                    if re.match(r'Test Cases?', match.group(1), re.IGNORECASE):
+                        # if there's a test case table, it's not a keyword file
+                        return False
+
+                    if (not found_keyword_table and
+                            re.match(r'(User )?Keywords?', match.group(1), re.IGNORECASE)):
+                        found_keyword_table = True
+        return found_keyword_table
+
+    def _post_request(self, session, endpoint, data):
```

Comment:
all http communication should be moved to separate class, like ApiClient with methods allowing for adding/deleting keywords and collections. These methods should ideally take objects of these classes as arguments: https://github.com/pbylicki/rfhub-new/blob/master/rfhub2/model/__init__.py based on the expected input model in api endpoint that they are calling.

Reply:
First part done, second part needs review